src/types/batch.rs

```diff
@@ -60,10 +60,9 @@ impl TryFrom<Bytes> for InboundData {
         }
         debug!("Parsing inbound data");
 
-        // Special-case a single request, rejecting invalid JSON.
-        if bytes.starts_with(b"{") {
-            let rv: &RawValue = serde_json::from_slice(bytes.as_ref())?;
-
+        // First, check if it's a single request
+        let rv: &RawValue = serde_json::from_slice(bytes.as_ref())?;
```

Comment:
this does mean batch requests are doing 2 linear passes, once for this `from_slice` and once for the `from_str`

we should be able to do that all in one pass

what if we did something like

```
enum DeserHelper<'a> {
    Batch(Vec<&'a RawValue>),
    Single(&'a RawValue)
}
```

Reply:
so this doesn't work because of https://github.com/serde-rs/serde/issues/1183 but that's fine, we can use the deserializer directly